### PR TITLE
logging_revamp: fix unittests logging leak SC-1530

### PR DESCRIPTION
**Pull request by CalvoM**
## Proposed Commit Message

logging_revamp: fix unittests logging leak

We are having some issues with logging leaks in unittests.

no-lp no-gh no-sc

## Test Steps

Run the all the tests `tox -e py310-test`.
Run the tests in test_util.py `tox -e py310-test uaclient/tests/test_util.py`.

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Review by CalvoM — COMMENTED** [review]
@lucasmoura @renanrodrigo I was able to solve the setup_logging mocking problem by mocking  `cfg.log_file` since it was the bone of contention.

File: uaclient/conftest.py [hunk 1]
```diff
@@ -85,6 +85,13 @@ def caplog_text(request):
     logs; if it returned caplog.text directly, that would always be empty.)
     """
     log_level = getattr(request, "param", logging.INFO)
```
Comment (lucasmoura) [inline uaclient/conftest.py:87]: It seems we don't need to create that variable anymore

**Review by lucasmoura — COMMENTED** [review]
Just to confirm, we call the `setup_logging` function on other scripts, i.e `upgrade_lts_contract.py`, where we don't pass `cfg.log_file` as a parameter. Would we have the isolation issue when testing those scripts ?

**Comment by CalvoM** [comment]
@lucasmoura We would have to mock the `UAConfig` object with the `FakeConfig` object, thus we will have no isolation issue. Even though it does not pass the `cfg.log_file` we default to it in `setup_logging` [here](https://github.com/canonical/ubuntu-pro-client/blob/43d519f413818073385039898246355ffac793f2/uaclient/cli.py#L1838)

**Comment by orndorffgrant** [comment]
@CalvoM just want to make sure you saw my latest questions :)

**Comment by CalvoM** [comment]
Oh! I updated the description but I forgot to update my comments. @orndorffgrant. For question 2, I have broken the commits into 2.

File: uaclient/conftest.py [hunk 2]
```diff
@@ -144,6 +154,10 @@ def __init__(
         ) -> None:
             if not cfg_overrides.get("data_dir"):
                 cfg_overrides.update({"data_dir": tmpdir.strpath})
+            if not cfg_overrides.get("log_file"):
+                cfg_overrides.update(
+                    {"log_file": tmpdir.join("log_file.log").strpath}
+                )
```
Comment (orndorffgrant) [inline uaclient/conftest.py:160]: I think this part is supposed to be in the second commit

**Review by orndorffgrant — APPROVED** [review]
Thanks for the explanation @CalvoM! I've verified that this works.

And thank you for splitting the commits! I just noticed one part of the diff that is currently in the first commit but should be in the second.

Pre-emptively approving but that should be fixed before merging.

File: uaclient/conftest.py (hunk 2 above)
Comment (CalvoM) [reply uaclient/conftest.py:160]: Updated @orndorffgrant .

Comment (orndorffgrant) [reply uaclient/conftest.py:160]: Thanks!

**Pull request merged**